module/zfs/dbuf.c

```diff
@@ -2243,10 +2242,8 @@ dbuf_undirty(dmu_buf_impl_t *db, dmu_tx_t *tx)
 	/*
 	 * If this buffer is not dirty, we're done.
 	 */
-	for (drp = &db->db_last_dirty; (dr = *drp) != NULL; drp = &dr->dr_next)
-		if (dr->dr_txg <= txg)
-			break;
-	if (dr == NULL || dr->dr_txg < txg)
+	dr = dbuf_find_dirty_eq(db, txg);
```

Comment:
Shouldn't this be `dbuf_find_dirty_lte` like the existing code to handle gaps in the txg sequence.

Reply:
No. The existing code just returns `B_FALSE` if `dr->dr_txg` is less than `txg`. This code retains those semantics.
```
-       for (drp = &db->db_last_dirty; (dr = *drp) != NULL; drp = &dr->dr_next)
-               if (dr->dr_txg <= txg)
-                       break;
-       if (dr == NULL || dr->dr_txg < txg)
+       dr = dbuf_find_dirty_eq(db, txg);
+       if (dr == NULL)
                return (B_FALSE);
-       ASSERT(dr->dr_txg == txg);
        ASSERT(dr->dr_dbuf == db);
```

Reply:
The only way you could reach the subsequent `ASSERT` is if the list isn't in descending order. In which case everything breaks.